### ISSUE TITLE
Validate Command Broken

### ISSUE THREAD

**Issue by sedouard** [Azure/azure-xplat-cli#2661]
CLI Version: **0.9.18**
OS Type: Linux Ubuntu
Installation via:  npm
Mode: ARM

Description:
`azure group template validate` fails with bad API response. Cannot read property 'providers' of undefined.

See:
https://travis-ci.org/Azure/azure-quickstart-templates/builds/117566373

Looks like the API response is bad here:
https://github.com/Azure/azure-xplat-cli/blob/v0.9.18-hotfix/lib/commands/arm/group/groupUtils._js#L312

Steps to reproduce:
1) Run `azure group create <group_name> location`
2) Then run `azure group template validate` on the elastic search template here:
https://github.com/Azure/azure-quickstart-templates/pull/1439
3) You'll get the same error as seen in the CI log.

Error stack trace:

**Please paste the content of ~/.azure/azure.err or C:\Users\username\\.azure\azure.err over here**
{ [TypeError: Cannot read property 'providers' of undefined]
  stack: [Getter/Setter],
  __frame: 
   { name: 'exports_validateTemplate__2',
     line: 107,
     file: '/app/node_modules/azure-scripty/node_modules/azure-cli/lib/commands/arm/group/groupUtils.js',
     prev: 
      { name: '__4',
        line: 161,
        file: '/app/node_modules/azure-scripty/node_modules/azure-cli/lib/commands/arm/group/group.template.js',
        prev: undefined,
        calls: 1,
        active: false,
        offset: 4,
        col: 17 },
     calls: 2,
     active: true,
     offset: 6,
     col: 33 },
  rawStack: [Getter] }
TypeError: Cannot read property 'providers' of undefined
  <<< async stack >>>
  at exports_validateTemplate__2 (/app/node_modules/azure-scripty/node_modules/azure-cli/lib/commands/arm/group/groupUtils.js:113:34)
  at __4 (/app/node_modules/azure-scripty/node_modules/azure-cli/lib/commands/arm/group/group.template.js:165:18)
  <<< raw stack >>>
    at getTemplateProviders (/app/node_modules/azure-scripty/node_modules/azure-cli/lib/commands/arm/group/groupUtils.js:312:46)
    at ___ (/app/node_modules/azure-scripty/node_modules/azure-cli/lib/commands/arm/group/groupUtils.js:118:38)
    at ___ (/app/node_modules/streamline/lib/callbacks/runtime.js:111:13)
    at /app/node_modules/azure-scripty/node_modules/azure-cli/lib/util/interaction.js:168:16
    at ___ (/app/node_modules/streamline/lib/callbacks/runtime.js:111:13)
    at /app/node_modules/azure-scripty/node_modules/azure-arm-resource/lib/resource/operations/deployments.js:1197:12
    at retryCallback (/app/node_modules/ms-rest/lib/filters/exponentialRetryPolicyFilter.js:140:9)
    at handleRedirect (/app/node_modules/ms-rest/lib/filters/redirectFilter.js:36:9)
    at /app/node_modules/azure-scripty/node_modules/azure-cli/lib/util/utils.js:522:7
    at handleRedirect (/app/node_modules/azure-scripty/node_modules/azure-cli/lib/util/utils.js:499:9)

**Comment by singhkay** [comment]
Tested a failing build here that gave this error with PowerShell and found the actual error returned - https://travis-ci.org/Azure/azure-quickstart-templates/builds/118407415

The error returned from PowerShell is which is correct
```
Code    : InvalidTemplate
Message : Deployment template validation failed: 'The provided value 'West US' for the template parameter 'location' at line '98' and column '22' is not valid. The parameter value is not 
          part of the allowed value(s): 'East US,South Central US,Central US,North Europe,West Europe,Southeast Asia,Japan West,Japan East,Australia Southeast,Australia East'.'.
```

**Comment by vivsriaus** [comment]
@sedouard This was fixed with the last release of CLI (0.9.19). Please update to the latest and let us know if you still run into issues. Thanks.
